src/leader_scheduler.rs

```diff
@@ -189,6 +189,17 @@ impl LeaderScheduler {
         }
     }
 
+    // Returns the last tick height for a given block index
+    pub fn max_tick_height_for_block(&self, block_index: u64) -> u64 {
```

Comment:
a leader rotation interval can span multiple blocks, is this function supposed to be called max_tick_height_for_slot() ?